lib/elixir_console_web/live/console_live/command_input_component.ex

```diff
@@ -24,8 +24,22 @@ defmodule ElixirConsoleWeb.ConsoleLive.CommandInputComponent do
      )}
   end
 
-  def handle_event("suggest", %{"key" => @tab_keycode, "value" => value}, socket) do
-    %{caret_position: caret_position, bindings: bindings} = socket.assigns
+  defp ensure_number(value) when is_number(value),
+    do: value
+
+  defp ensure_number(value), do: String.to_integer(value)
+
+  def handle_event(
+        "suggest",
+        %{"key" => @tab_keycode, "value" => value, "caret_position" => caret_position},
+        socket
+      ) do
+    %{bindings: bindings} = socket.assigns
+
+    # When testing this event using render_keydown/up, even if the metadata is defined as a number,
+    # we're receiving the value here as a string.
+    # This happens only in tests though, when running the server we correctly receive it as a number.
+    caret_position = ensure_number(caret_position)
```

Comment:
Possibly due to a bug or limitation of Phoenix.LiveViewTest?

Reply:
Not sure if bug or intentional - apparently this is the line where it's converted https://github.com/phoenixframework/phoenix_live_view/blob/32fcf054897fa3c21c9fa57dceee48fb58b60000/lib/phoenix_live_view/test/client_proxy.ex#L190.

Reply:
I think this is the commit (https://github.com/phoenixframework/phoenix_live_view/commit/fa4df865d8905a3fa84f803d3ec598a5cd86a655#diff-39d73afd0f8def95e8eb610b5fa70619ff5a6755f2445aeea3d74d82a30f46a2R368-R478) that started stringify'ing, since `v0.1.0`, so sounds like intentional...

Reply:
Nice find! Even if intentional, it's not the same behavior you get when testing the server so maybe it's something to bring up in the Elixir forum?